src/sentry/data_export/processors/discover.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import absolute_import
+
+import logging
+
+from sentry.api.utils import get_date_range_from_params
+from sentry.snuba import discover
+from sentry.models import Project
+
+from ..base import ExportError
+
+logger = logging.getLogger(__name__)
+
+
+ALIAS_FIELDS = [("user", ["user.name", "user.email", "user.username", "user.ip"])]
+
+
+class DiscoverProcessor(object):
+    """
+    Processor for exports of discover data based on a provided query
+    """
+
+    def __init__(self, organization_id, discover_query):
+        self.projects = self.get_projects(discover_query["project"])
+        self.start, self.end = get_date_range_from_params(discover_query)
+        self.params = {
+            "organization_id": organization_id,
+            "project_id": [project.id for project in self.projects],
+            "start": self.start,
+            "end": self.end,
+        }
+        self.header_fields = discover_query["field"]
+        self.data_fn = self.get_data_fn(
+            fields=discover_query["field"], query=discover_query["query"], params=self.params
+        )
+
+    @staticmethod
+    def get_projects(project_ids):
+        try:
+            if isinstance(project_ids, list):
+                return Project.objects.filter(id__in=project_ids)
+            else:
+                return [Project.objects.get_from_cache(id=project_ids)]
+        except Project.DoesNotExist:
+            raise ExportError("Requested project does not exist")
+
+    @staticmethod
+    def get_data_fn(fields, query, params):
+        def data_fn(offset, limit):
+            return discover.query(
+                selected_columns=fields,
+                query=query,
+                params=params,
+                offset=offset,
+                limit=limit,
+                referrer="api.organization-events-v2",
+                auto_fields=True,
+                use_aggregate_conditions=True,
+            )
+
+        return data_fn
+
+    def alias_fields(self, raw_data):
```

Comment:
If this function was exposed by the `discover` module you might not need to redefine the alias fields again.

Reply:
Are you talking about using a similar function that might be in `src/sentry/snuba/discover.py`?

Reply:
Are you talking about using a similar function that might be in `src/sentry/snuba/discover.py`?

Reply:
Yes. That module already has access to the field alias composition and would be a good place to do the coalesce. We'd still need all the component fields in API responses though as the UI needs to do its own coalesce :cry:

Reply:
I've been picking at this a few days, and I don't think I can directly use the aliasing from (`FIELD_ALIASES`) in `event_search.py` since I need the exported CSV to match what is shown in the browser

Reply:
I'm going to take a look at replicating some of the stuff in `fieldRenders.tsx` instead, but let me know if this is the wrong approach

Reply:
Reproducing the fieldRenderer code is a good plan.